### PR TITLE
Pm 29 create shelter shelter id part 2 (edit and delete)

### PR DESCRIPTION
**Pull request by dotte-osu**
Added "Edit" and "Delete" shelter functions.

Edit part:
image

I didn't add password change because that is more like shelter "account" data change rather than shelter "information" data. Now that I think about it.... I am not sure if we should use this "login email address" as their shelter "contact email address". (sorry, I am not sure how I should describe this...let me know if this is confusing.) Maybe we should have an optional "contact email address" column in the database?...but that also seems confusing. Maybe it's okay as it is.

Delete part:
If a shelter wants to delete their shelter data, the page will give them a warning pop up. 
image
Once the shelter click `OK`, the page will take them to the home page and the shelter is automatically logged out.

### CODE REVIEW

File: src/components/CreateAccountForm/CreateAccountForm.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import AuthContext from '../../context/AuthContext';
 import UsersService from '../../services/usersService';
 import SheltersService from '../../services/sheltersService';
 import PasswordChecklist from 'react-password-checklist'
-import USStates from './USStates.json'
+import USStates from '../../config/USStates.json'
```
Comment (dotte-osu) [inline src/components/CreateAccountForm/CreateAccountForm.js]: moved the json file to config folder

File: src/routes/SheltersPege/SheltersPege.js [hunk 2]
```diff
@@ -21,35 +27,52 @@ export default class SheltersPege extends Component {
...
+      state: null,
+      zip: null,
+      email: null,
+      phoneNumber: null,
+      website: null,
+      error: null
     };
     this.inputChanged = this.inputChanged.bind(this);
+    this.onClickEdit = this.onClickEdit.bind(this);
+    this.onClickUpdate = this.onClickUpdate.bind(this);
```
Comment (dotte-osu) [inline src/routes/SheltersPege/SheltersPege.js]: This actually update the database

File: src/routes/SheltersPege/SheltersPege.js [hunk 3]
```diff
@@ -21,35 +27,52 @@ export default class SheltersPege extends Component {
...
+      city: null,
+      state: null,
+      zip: null,
+      email: null,
+      phoneNumber: null,
+      website: null,
+      error: null
     };
     this.inputChanged = this.inputChanged.bind(this);
+    this.onClickEdit = this.onClickEdit.bind(this);
```
Comment (dotte-osu) [inline src/routes/SheltersPege/SheltersPege.js]: This switches the "view" mode and "edit" mode

File: src/components/CreateAccountForm/CreateAccountForm.js (hunk 1 above)
Comment (xHoudek) [reply src/components/CreateAccountForm/CreateAccountForm.js]: Moving it is a good idea! Do you think it should go into utils instead? It isn't really something that is configurable

File: src/routes/SheltersPege/SheltersPege.js (hunk 3 above)
Comment (xHoudek) [reply src/routes/SheltersPege/SheltersPege.js]: I think it would be more clear if we changed this name. Maybe something like `handleEditButtonClick`? Adding code comments above the two functions could also be good for clarification

File: src/routes/SheltersPege/SheltersPege.js [hunk 4]
```diff
@@ -1,15 +1,21 @@
...
 import SheltersService from '../../services/sheltersService';
 import * as logUtils from '../../components/Utils/Logger';
 import PetList from '../../components/PetList';
 import PetsService from '../../services/petsService';
 import { Link } from 'react-router-dom';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import USStates from '../../config/USStates.json'
 var log = logUtils.getLogger()
 
 export default class SheltersPege extends Component {
```
Comment (xHoudek) [inline src/routes/SheltersPege/SheltersPege.js]: Not part of this PR, but something I noticed: This should be `SheltersPage` instead of `SheltersPege`. Same with the filename

File: src/routes/SheltersPege/SheltersPege.js [hunk 5]
```diff
@@ -77,34 +100,210 @@ export default class SheltersPege extends Component {
     return filters;
   }
 
+  updateState(shelter) {
```
Comment (xHoudek) [inline src/routes/SheltersPege/SheltersPege.js]: `this.setState` can take an object with multiple attributes, so you probably only need one call to `this.setState`

```
this.setState({
  shelterName: shelter[0].ShelterName,
  address: shelter[0].Address.substring(0, shelter[0].Address.indexOf(',')),
  city: shelter[0].Address.substring(shelter[0].Address.indexOf(',') + 2, shelter[0].Address.lastIndexOf(',')),
  state: shelter[0].Address.substring(shelter[0].Address.lastIndexOf(',') + 2, shelter[0].Address.lastIndexOf(',') + 4),
  zip: parseInt(shelter[0].Address.substring(shelter[0].Address.length - 5)),
  email: shelter[0].EmailAddress,
  phoneNumber: parseInt(shelter[0].PhoneNumber),
  website: shelter[0].Website
});
```

File: src/routes/SheltersPege/SheltersPege.js [hunk 6]
```diff
@@ -21,35 +27,52 @@ export default class SheltersPege extends Component {
...
     SheltersService.getShelter(shelterID)
-      .then(shelter => this.setState({ shelter: shelter }))
-      .catch(error => console.log(error));
+      .then(shelter => {
+        this.setState({ shelter: shelter })
+        this.updateState(shelter)
+      })
+      .catch(error => {
+        log.error(error)
+        this.setState({ error: "No shelter data avairable" });
```
Comment (xHoudek) [inline src/routes/SheltersPege/SheltersPege.js]: Small typo: should be "available"

File: src/routes/SheltersPege/SheltersPege.js [hunk 7]
```diff
@@ -77,34 +100,210 @@ export default class SheltersPege extends Component {
...
+      phoneNumber,
+      website
+    } = this.state;
+
+    let data = {
+      "shelterID": parseInt(shelterID),
+      "shelterName": shelterName,
+      "address": address + ", " + city + ", " + state + " " + zip,
+      "emailAddress": email,
+      "phoneNumber": parseInt(phoneNumber),
```
Comment (xHoudek) [inline src/routes/SheltersPege/SheltersPege.js]: Is `parseInt()` necessary? I thought the DB was now storing this value as a string

File: src/routes/SheltersPege/SheltersPege.js [hunk 8]
```diff
@@ -77,34 +100,210 @@ export default class SheltersPege extends Component {
...
-              </tr>
-            </tbody>
-          </table>
-        </div>
+      </div>
+    );
+  }
+
+  renderUSStates() {
+    let stateOptions = [<options></options>];
```
Comment (xHoudek) [inline src/routes/SheltersPege/SheltersPege.js]: `<options></options>` can be removed. I added that in there before the states were moved to a JSON, where there is an empty string built in. Also the `<options>` tag doesn't exist, which is totally my bad lol

File: src/services/sheltersService.js [hunk 9]
```diff
@@ -29,6 +30,35 @@ const SheltersService = {
       (!res.ok)
         ? res.json().then(e => Promise.reject(e))
         : res.json());
+  },
+  updateShelter(shelter) {
+    console.log("postShelter")
```
Comment (xHoudek) [inline src/services/sheltersService.js]: console logs can be removed

File: src/services/sheltersService.js [hunk 10]
```diff
@@ -29,6 +30,35 @@ const SheltersService = {
...
+    console.log(shelter.shelterID + ".")
+    console.log(shelter)
+    return fetch(`${HOSTNAME}/shelters/${shelter.shelterID}`, {
+      method: 'PATCH',
+      headers: {
+        'content-type': 'application/json',
+        'authorization': AuthService.getToken()
+      },
+      body: JSON.stringify({ ...shelter })
```
Comment (xHoudek) [inline src/services/sheltersService.js]: This is a question for my own edification. Why use `{ ...shelter}` instead of `shelter`? I still have yet to fully grasp the usage of `...` in JS

File: src/routes/SheltersPege/SheltersPege.js [hunk 11]
```diff
@@ -77,34 +100,210 @@ export default class SheltersPege extends Component {
...
+                    onChange={e => this.inputChanged('website', e.target.value)} /></td>
+                </tr>
+              </tbody>
+            </table>
+            <hr />
+          </div>
+          <div className="row">
+            <div className="col-sm-12 text-center">
+              <button className="btn btn-primary btn-md m-2" onClick={this.onClickUpdate}>Update</button>
+              <button className="btn btn-success  btn-md m-2" onClick={this.onClickEdit}>Cancel</button>
```
Comment (xHoudek) [inline src/routes/SheltersPege/SheltersPege.js]: I think this should be changed to `className="btn btn-outline-primary"`

**Review by xHoudek — CHANGES_REQUESTED** [review]
I'm struggling to get this to function properly when I test. I log in as a shelter and am able to access the edit page, but when I change something and click "Update", nothing happens. It looks like all of the information stays the same and the information doesn't actually get updated. Not sure if that is something on my end or what.

Also, every time I type a new character in on of the edit inputs, a new GET request is made. I think we should change it so that GET request isn't being made every time.

Overall though, I really like they layout and the way the edit section fits into the shelter page! Great design. Just some kinks to iron out.

Delete functionality seems to work as expected. Pets associated with the shelter are also removed, which is great!

File: src/components/CreateAccountForm/CreateAccountForm.js (hunk 1 above)
Comment (dotte-osu) [reply src/components/CreateAccountForm/CreateAccountForm.js]: ah! yeah utils is better! I will move it.

**Comment by xHoudek** [comment]
After some discussion, it looks like the unable to update issue may be on my end (most likely is). @asching7108 could you test this, along with [the backend PR](https://github.com/asching7108/purrfect-match-api/pull/30), to see if you're seeing the same issue I am?

File: src/routes/SheltersPege/SheltersPege.js (hunk 3 above)
Comment (dotte-osu) [reply src/routes/SheltersPege/SheltersPege.js]: fixed!

File: src/routes/SheltersPege/SheltersPege.js (hunk 4 above)
Comment (dotte-osu) [reply src/routes/SheltersPege/SheltersPege.js]: fixed

File: src/routes/SheltersPege/SheltersPege.js (hunk 5 above)
Comment (dotte-osu) [reply src/routes/SheltersPege/SheltersPege.js]: ah! pretty! thank you!

File: src/routes/SheltersPege/SheltersPege.js (hunk 7 above)
Comment (dotte-osu) [reply src/routes/SheltersPege/SheltersPege.js]: good catch.

File: src/routes/SheltersPege/SheltersPege.js (hunk 8 above)
Comment (dotte-osu) [reply src/routes/SheltersPege/SheltersPege.js]: fixed

File: src/services/sheltersService.js (hunk 9 above)
Comment (dotte-osu) [reply src/services/sheltersService.js]: removed

File: src/services/sheltersService.js (hunk 10 above)
Comment (dotte-osu) [reply src/services/sheltersService.js]: In this case, ... formats an array to json nicely (I believe but I might be wrong!), but I am simply passing a json object so I removed it!

File: src/routes/SheltersPege/SheltersPege.js (hunk 11 above)
Comment (dotte-osu) [reply src/routes/SheltersPege/SheltersPege.js]: much prettier! thank you!

**Comment by dotte-osu** [comment]
Addressed the code comments. Let me know if the code doesn't work.

File: src/routes/SheltersPege/SheltersPege.js [hunk 12]
```diff
@@ -1,14 +1,20 @@
...
 import * as logUtils from '../../components/Utils/Logger';
 import PetList from '../../components/PetList';
 import PetsService from '../../services/petsService';
 import { Link } from 'react-router-dom';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import USStates from '../../components/Utils/USStates.json'
 var log = logUtils.getLogger()
 
-export default class SheltersPege extends Component {
+export default class SheltersPage extends Component {
```
Comment (xHoudek) [inline src/routes/SheltersPege/SheltersPege.js]: I think the folder and Files need to be renamed to say "SheltersPage" too. Same with the .css file

File: src/routes/SheltersPege/SheltersPege.js (hunk 7 above)
Comment (xHoudek) [reply src/routes/SheltersPege/SheltersPege.js]: Looks like this wasn't updated

**Review by xHoudek — CHANGES_REQUESTED** [review]
Looks like there are a few more minor typos and things to change. I am still unable to update anything, just waiting for Esther to test

**Comment by dotte-osu** [comment]
Sorry, fixed all the "SheltersPage" issues. I also re-clone the repo and tested in Chrome and Firefox. The update function works fine. If `SheltersService.updateShelter` is called, the issue must be in the backend. Most likely update sql query because I updated the phone number data type to varchar(10) and the error message should show up in the backend console.

File: src/routes/SheltersPage/SheltersPage.js [hunk 13]
```diff
@@ -0,0 +1,380 @@
...
+    this.setState({ shelterID });
+
+    SheltersService.getShelter(shelterID)
+      .then(shelter => {
+        this.setState({ shelter: shelter })
+        this.updateState(shelter)
+      })
+      .catch(error => {
+        log.error(error)
+        this.setState({ error: "No shelter data avairable" });
```
Comment (xHoudek) [inline src/routes/SheltersPage/SheltersPage.js]: This typo still needs to be fixed I think

**Review by xHoudek — APPROVED** [review]
Great job on this! Glad we were able to iron out the kinks
